### ISSUE TITLE
My profile: "My accounts" - some accounts show old votes from 2-3 election cycles ago that have already been recovered

### ISSUE THREAD

**Issue by mochet** [Joystream/pioneer#4668]
# Problem
SSIA
I don't have a screenshot because I recovered the votes via the "Past Votes" page in the end which managed to rectify this behavior.

**Comment by dmtrjsg** [comment]
cc @vrrayz  @chrlschwb

**Comment by thesan** [comment]
@mochet is this still happening ?

**Comment by mochet** [comment]
> @mochet is this still happening ?

Yes, I just checked and one of my accounts is showing a vote from January 13th 2023. Another is showing a vote from November 2023. So there isn't really any consistency with the bug.

I also checked on https://pioneer.joyutils.org/#/profile (an alternative hosted version of Pioneer which is the same as the current version of Pioneer AFAIK just with different QN/RPC nodes) and it also has the same bug. 

I should also note that since the `past votes` page bug was fixed (https://github.com/Joystream/pioneer/issues/4667) this is much less of an issue as a user can just recover their vote stake using that page properly now.

**Comment by traumschule** [comment]
Task: Make sure only the last vote per staking account is shown.
